vertx-web-graphql/src/main/java/io/vertx/ext/web/handler/graphql/impl/ApolloWSHandlerImpl.java

```diff
@@ -115,6 +121,10 @@ public void handle(RoutingContext routingContext) {
   private void handleConnection(RoutingContext routingContext, ServerWebSocket serverWebSocket) {
     Map<String, Subscription> subscriptions = new ConcurrentHashMap<>();
 
+    if (connectionHandler != null) {
```

Comment:
What do you think of merging connectionHandler and endHandler with messageHandler and add two more types (SOCKET_CREATED, SOCKET_DESTROYED) ?

Reply:
I think connection and end handlers have different purpose than message handler. Connection and disconnection messages are handled by the message handler, but connection and disconnection “action” are handled by these handlers. The information we have is different.

Reply:
@zenios good catch

@rogelio-o please do not update the PR, I'm merging it. I will update this part of the code.